### PR TITLE
Network Offline Functionality and Data Correctness

### PR DESCRIPTION
**Pull request by neelp1030**
Add a NetworkMonitor that uses callbacks to determine whether to contact Firestore or the local cache for GET queries. All the repositories now use this NetworkMonitor to determine the source to contact for their GET queries. Also remove await() for write/update/modify queries in the repositories wherever possible, so that when the app is offline, to the user it seems like the changes took effect seamlessly. Ensure data correctness, specifically inventory, when multiple users (all offline) harvest and/or sell and then go online roughly around the same time.

### CODE REVIEW

File: app/src/main/java/com/example/farmeraid/data/InventoryRepository.kt [hunk 1]
```diff
@@ -3,20 +3,21 @@ package com.example.farmeraid.data
...
 import com.google.firebase.firestore.FieldValue
 import com.google.firebase.firestore.FirebaseFirestore
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.tasks.await
-import javax.inject.Inject
 
 // TODO: currently, we have mock demo functionality but need to modify to use firestore db after demo
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/data/InventoryRepository.kt:15]: Lets remove this comments

File: app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt [hunk 2]
```diff
@@ -1,39 +1,35 @@
...
-import java.time.Instant
-import java.time.LocalDate
 import java.time.LocalDateTime
-import java.time.LocalTime
-import java.time.ZoneId
-import java.time.temporal.TemporalAdjuster
-import java.time.temporal.TemporalAdjusters
-import java.util.Date
 
 // TODO: currently, we have mock demo functionality but need to modify to use firestore db after demo
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt:18]: These too

File: app/src/main/java/com/example/farmeraid/data/source/NetworkMonitor.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import androidx.core.content.ContextCompat.getSystemService
+import com.google.firebase.firestore.Source
+import kotlinx.coroutines.flow.MutableStateFlow
+
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/data/source/NetworkMonitor.kt:11]: Add refence where we got this code

**Review by RishanR — APPROVED** [review]
LGTM! Good work with offline mode, this is HUUUUGE!

**Review by adshayanB — APPROVED** [review]
Other then the nits this is great! Our conversation today really helped push this development! 

Good work!

**Pull request merged**